tests/kernel/stack/stack/src/test_stack_contexts.c

```diff
@@ -179,6 +183,99 @@ void test_stack_alloc_thread2thread(void)
 			"resource pool is smaller then requested buffer");
 }
 
+static void low_prio_wait_for_stack(void *p1, void *p2, void *p3)
+{
+	struct k_stack *pstack = p1;
+	stack_data_t output;
+
+	k_stack_pop(pstack, &output, K_FOREVER);
+	zassert_true(output == 0xccc,
+	    "The low priority thread get the stack data failed lastly");
+}
+
+static void high_prio_t1_wait_for_stack(void *p1, void *p2, void *p3)
+{
+	struct k_stack *pstack = p1;
+	stack_data_t output;
+
+	k_stack_pop(pstack, &output, K_FOREVER);
+	zassert_true(output == 0xaaa,
+	    "The highest priority and waited longest get the stack data failed firstly");
+}
+
+static void high_prio_t2_wait_for_stack(void *p1, void *p2, void *p3)
+{
+	struct k_stack *pstack = p1;
+	stack_data_t output;
+
+	k_stack_pop(pstack, &output, K_FOREVER);
+	zassert_true(output == 0xbbb,
+	   "The higher priority and waited longer get the stack data failed secondly");
+}
+
+/**
+ * @brief Test multi-threads to get data from stack.
+ *
+ * @details Define three threads, and set a higher priority for two of them,
+ * and set a lower priority for the last one. Then Add a delay between
+ * creating the two high priority threads.
+ * Test point:
+ * 1. Any number of threads may wait(K_FOREVER set) on an empty stack
+ * simultaneously.
+ * 2. When data is pushed, it is given to the highest priority
+ * thread that has waited longest.
+ *
+ * @ingroup kernel_stack_tests
+ */
+void test_stack_multithread_competition(void)
+{
+	int old_prio = k_thread_priority_get(k_current_get());
+	int prio = 10;
+	stack_data_t test_data[3];
+
+	memset(test_data, 0, sizeof(test_data));
+	k_thread_priority_set(k_current_get(), prio);
+
+	/* Set up some values */
+	test_data[0] = 0xAAA;
+	test_data[1] = 0xBBB;
+	test_data[2] = 0xCCC;
+
```

Comment:
I think we should define three macroes for "0xaaa", "0xbbb" and "0xccc". Then, replace the hard codes at lines 192, 202, and 211 with them.

Reply:
OK, no problem.

Reply:
Hi @LeiW000 , I had updated.